docs/root/intro/arch_overview/ssl.rst

```diff
@@ -158,3 +158,22 @@ infrastructure.
 
 Client TLS authentication filter :ref:`configuration reference
 <config_network_filters_client_ssl_auth>`.
+
+.. _arch_overview_ssl_trouble_shooting:
+
+Trouble shooting
+----------------
+
+When Envoy originates TLS when making connections to upstream clusters, its error will be logged into
+:ref:`UPSTREAM_TRANSPORT_FAILURE_REASON<config_access_log_format_upstream_transport_failure_reason>` field.
+Common errors are:
+
+* ``Secret is not delivered via SDS``: Envoy is still waiting SDS to deliver key/cert or root CA.
```

Comment:
nit: I think I would call this error "Secret not supplied by SDS" WDYT?
nit: "still waiting for"